zerver/webhooks/circleci/view.py

```diff
@@ -23,133 +48,98 @@ def api_circleci_webhook(
-    for commit in payload["all_commit_details"]:
-        author_name = commit["author_name"].tame(check_string)
-        author_username = commit["author_login"].tame(check_none_or(check_string))
-        if author_username is not None:
-            author_names.add(f"{author_name} ({author_username})")
-        else:
-            author_names.add(author_name)
-
-        if commit["committer_email"].tame(check_none_or(check_string)) is not None:
-            committer_name = commit["committer_name"].tame(check_string)
-            committer_username = commit["committer_login"].tame(check_none_or(check_string))
-            if committer_username is not None:
-                committer_names.add(f"{committer_name} ({committer_username})")
-            else:
-                committer_names.add(committer_name)
-
-    author_names_list = list(author_names)
-    author_names_list.sort()
-    committer_names_list = list(committer_names)
-    committer_names_list.sort()
-    authors = ", ".join(author_names_list)
-    committers = ", ".join(committer_names_list)
-
-    # Add the authors' information to the body.
-    if len(author_names_list) > 1:
-        body += f"- **Authors:** {authors}"
-    else:
-        body += f"- **Author:** {authors}"
+    return payload["project"]["name"].tame(check_string)
+
 
-    # Add information about the committers if it was provided.
-    if len(committer_names) > 0:
-        if len(committer_names) > 1:
-            body += f"\n- **Committers:** {committers}"
-        else:
-            body += f"\n- **Committer:** {committers}"
+def get_commit_details(payload: WildValue) -> str:
+    if "vcs" in payload["pipeline"]:  # GitHub and BitBucket associated pipelines.
+        commit_subject = payload["pipeline"]["vcs"]["commit"]["subject"].tame(check_string)
+        commit_id = payload["pipeline"]["vcs"]["revision"].tame(check_string)[:7]
+        commit_details = f"{commit_id}: {commit_subject}"
 
-    return body
+        author_name = payload["pipeline"]["vcs"]["commit"]["author"]["name"].tame(check_string)
 
+        return COMMIT_INFO_TEMPLATE.format(
+            commit_details=commit_details,
+            author_name=author_name,
+        )
 
-def super_minimal_body(payload: WildValue) -> str:
-    branch_name = payload["branch"].tame(check_string)
-    status = payload["status"].tame(check_string)
-    formatted_status = outcome_to_formatted_status_map.get(status, status)
-    build_url = payload["build_url"].tame(check_string)
-    username = payload["username"].tame(check_string)
-    return f"[Build]({build_url}) triggered by {username} on branch `{branch_name}` {formatted_status}."
+    else:  # Other providers (GitLab).
+        commit_title = payload["pipeline"]["trigger_parameters"]["gitlab"]["commit_title"].tame(
+            check_string
+        )
+        commit_id = payload["pipeline"]["trigger_parameters"]["gitlab"]["checkout_sha"].tame(
+            check_string
+        )[:7]
```

Comment:
Why 7? Generally we tend to go with 12 or so, since some larger projects like Linux require more than 7 characters of the hash to be unique.

Reply:
Opened https://github.com/zulip/zulip/issues/23475 for this follow-up.

Reply:
I sliced the SHA to 7 characters to maintain consistency with GitHub's interface, which displays a shortened SHA of 7 characters. That's probably why the other integrations also slice it to 7 characters.